### ISSUE TITLE
CRA Express 2.0.0

### ISSUE THREAD

**Issue by antonybudianto** [antonybudianto/create-react-app-express#7]
## New features:
- **Loaders**: functions that take express app instance and add the needed middleware / setup according to its usage.
   - Included basic loaders:
     - `@cra-express/static-loader` (for handling static assets)
     - `@cra-express/universal-loader` (for universal rendering)
       - this loader should be called **last**
   - If you use `createReactAppExpress` API, it used both loaders by default
   - Usecases / background:
      - useful when you want to build your own routes on top of `create-react-app-express` middleware. currently all routes are consumed by universal loader so it's not possible to create your own routes (e.g.: API routes, etc)
      - able to use universal rendering for existing/managed express app.
   - the naming can be confused with webpack loaders, any suggestions are welcome
- Babel bundling (for non-es6 compatibility)

## Plan:
To make it more middleware oriented, than all-in-one magic API. Loaders are still considered magic though, but at least they only do **one** thing, unlike previous API.
- Forkable/custom loaders
   - Each loader will have their own package
   - The loaders source code will be still in this repo (mono-repo), will use Lerna
   - Developer will have more flexibility to fork and customize their own loader
   - Since loaders are just functions, you can write your own, publish, and replace the built-ins easily

## Breaking changes:
- new package name: create-react-app-express -> @cra-express/{core, static-loader, universal-loader}
- `createReactAppExpress()` in on core, and it keeps the compatibility
- in short, you just need to update your package name and it should be okay

## Pre-release
You can try by installing:
```
npm i @cra-express/core@0.1.1
```
> ⚠️ **NOT FOR PRODUCTION USAGE**
